### DISCUSSION TITLE
Type Hinting / Autocomplete / Function Definitions with VSCode

### DISCUSSION THREAD

**Discussion by pep-sanwer** [ibis-project/ibis#10271] [Q&A]
Would anyone have insight into how to get type hinting / autocomplete working with VSCode?
I'm bizarrely seeing type hinting / autocomplete only when using an Jupyter interactive session or if I directly type hint objects, but not in the normal editor mode.

- Default language server (Pylance)
- Backend: `snowflake`
- Seeing IDE help (type hints, autocomplete, etc.) as expected with all other packages

Ex:
```python
import ibis

con = ibis.backend.snowflake.from_connection(...)
```
![image](https://github.com/user-attachments/assets/6a77d2dd-c940-401c-bf0c-c12a6f2ea4a5)

This works as expected
```python
import ibis
import ibis.backends.snowflake

_: ibis.backends.snowflake.Backend = ibis.snowflake
con = _.from_connection(...)
con.list_tables()
```
![image](https://github.com/user-attachments/assets/80836716-d7ab-433c-9ace-333edbf3d150)

And interactive (Jupyter) session works as well:
![image](https://github.com/user-attachments/assets/5d8558ee-c0b2-41a0-a0b2-498f26ebecd0)

Any help would be greatly appreciated!

**Comment by hershelm** [comment]
+1 having the same issue. ideally, i would like to see hints / auto-complete working in interactive mode in VS Code notebooks. so i could see things like functions available (filter, mutate, etc.), and column names in the tables when using the `_.COLUMN_NAME` convention within the pipelines. is this possible?

**Reply by dschneiderch** [reply]
i actually have the opposite issue. 
`con.` autocompletes to list_tables()
but once I type `list_tables()` and keep my cursor inside the parens I don't see the docstring.
